src/main.ts

```diff
@@ -95,100 +102,168 @@ export default class AdvancedURI extends Plugin {
         });
 
 
-        this.registerObsidianProtocolHandler("advanced-uri", async (e) => {
-            const parameters = e as unknown as Parameters;
+        this.registerObsidianProtocolHandler("advanced-uri", this.handleUriAction);
```

Comment:
The original callback here is moved into the below `handleUriAction` so that it can be called both by the protocol handler as well as the HTTP request listener.